scripts/setup-branch-protection.sh

```
@@ -0,0 +1,196 @@
+
+# Create main branch if it doesn't exist
+ensure_main_branch() {
+    print_status "Checking if main branch exists..."
+
+    if GH_PAGER=cat gh api "repos/${REPO}/branches/main" &> /dev/null; then
+        print_status "Main branch already exists"
+    else
+        print_warning "Main branch doesn't exist. Creating it from develop..."
+
+        # Get the SHA of the develop branch
+        DEVELOP_SHA=$(GH_PAGER=cat gh api "repos/${REPO}/git/refs/heads/develop" | jq -r '.object.sha')
```

Shellcheck (0.10.0)

[warning] Use var=$(command) to assign output (or quote to assign string).

(SC2209)